smpplib/gsm.py

```diff
@@ -10,19 +10,21 @@ def make_parts(text, encoding=consts.SMPP_ENCODING_DEFAULT):
     """Returns tuple(parts, encoding, esm_class)"""
     try:
         # Try to encode with the user-defined encoding first.
-        encode, split_size, part_size = ENCODINGS[encoding]
+        encode, split_length, part_size = ENCODINGS[encoding]
         encoded_text = encode(text)
     except KeyError:
         raise NotImplementedError('encoding is not supported: %s' % encoding)
     except UnicodeError:
         # Fallback to UCS-2.
         encoding = consts.SMPP_ENCODING_ISO10646
-        encode, split_size, part_size = ENCODINGS[encoding]
+        encode, split_length, part_size = ENCODINGS[encoding]
         encoded_text = encode(text)
 
-    if len(text) > split_size:
+    if len(text) > split_length:
         # Split the text into well-formed parts.
         esm_class = consts.SMPP_GSMFEAT_UDHI
+        # FIXME: 7-bit encoding has variable-length characters.
+        # FIXME: it means that a character may be broken by splitting.
```

Comment:
Just had a thought that there's another issue too. We check `len(text)` but later we split encoded byte-string. It means that 2-bytes characters are encountered as they were 1-octet characters.